api/s2n.h

```diff
@@ -241,6 +241,22 @@ S2N_API extern int s2n_init(void);
  */
 S2N_API extern int s2n_cleanup(void);
 
+/**
+ * Determines whether s2n-tls is in FIPS mode.
+ *
+ * s2n-tls enters FIPS mode on initialization when the linked libcrypto has FIPS mode enabled. Some
+ * libcryptos, such as AWS-LC-FIPS, have FIPS mode enabled by default. With other libcryptos, such
+ * as OpenSSL, FIPS mode must be enabled before initialization by calling `FIPS_mode_set()`.
+ *
+ * s2n-tls MUST be linked to a FIPS libcrypto and MUST be in FIPS mode in order to comply with FIPS
+ * requirements. Applications desiring FIPS compliance should use this API to ensure that s2n-tls
+ * has been properly linked with a FIPS libcrypto and has successfully entered FIPS mode.
+ *
+ * @param fips Set to true if s2n-tls is in FIPS mode, set to false otherwise.
+ * @returns S2N_SUCCESS on success. S2N_FAILURE on failure.
+ */
+S2N_API extern int s2n_is_fips(bool *fips);
+
```

Comment:
And we're definitely sure we're never going to want to return anything other than true/false here?

Reply:
I think so. I'm not able to think of another value that might make sense to return here, but let me know if you disagree.

Also, even if we made this an enum, I can imagine applications asserting a specific value to ensure FIPS compliance:
```
s2n_fips_mode fips_mode = S2N_FIPS_MODE_DISABLED;
s2n_get_fips_mode(&fips_mode);
if (fips_mode != S2N_FIPS_MODE_ENABLED) {
    exit(1);
}
```

So even if we made this an enum, it seems likely that we'd want a separate API for the new value anyway in order to avoid breaking FIPS compliance checks like this.

Reply:
Although I guess this assumes that our new value would somehow indicate compliance. If we wanted to add a new value to indicate non-compliance it seems like we might be able to do that safely.

Reply:
We discussed this offline: using an enum here won't really be any less usable, and allows us to add a new value in the future if we decide to. So I will change the argument to an enum.